### PR TITLE
feat: add oracle linux

### LINKED ISSUE #671: Add support for Oracle Enterprise Linux
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Description

A nice feature for this project would be the addition of Oracle Enterprise Linux. 

### Use Case(s)

The possibility to create Oracle Enterprise Linux 8 Templates

### Potential Configuration

This is discussed in this pull request: https://github.com/vmware-samples/packer-examples-for-vsphere/pull/670

### References

_No response_

### PR DESCRIPTION
**Pull request by virtualistic**
# 

## Summary of Pull Request

As I requested here:
https://github.com/vmware-samples/packer-examples-for-vsphere/discussions/659 and inspired by your suggestion, created a new build for Oracle Linux 8.  Had to change the menu a bit to "make room" for the Oracle 8 template at position 19. (I skipped position 18 for Oracle 9 in the future). I was able to successfully build a new Oracle 8 template but hopefully you have some time to test and verify this.

## Type of Pull Request

- [ ] This is a bugfix. `type/bug`
- [X] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.
      Please describe:

## Related to Existing Issues

Closes #671

## Test and Documentation Coverage

- [X] Tests have been completed.
- [ ] Documentation has been added or updated.

## Breaking Changes?

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

**Review by tenthirtyam — COMMENTED** [review]
The build.yaml needs to be updated. This is used to generated the build.sh with gomplate.

The README.md would also need an update.

**Comment by tenthirtyam** [comment]
Marked as drafted until the CLA is approved.

**Comment by virtualistic** [comment]
I added a new commit which added Oracle Linux as an entry to the README
Also added an entry for build.yaml

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Hi @virtualistic -

Please run the following to generate the `build.sh`:

```bash
gomplate -c build.yaml -f build.tmpl -o build.sh
```

Additionally, in the `README.md` and update is needed to list the download information for each the distribution:

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by virtualistic** [comment]
Created a new commit that:
- Rebuilt the build.sh using the command:
`gomplate -c build.yaml -f build.tmpl -o build.sh`

- Added the source for the Oracle ISO to the README

**Comment by tenthirtyam** [comment]
Thanks! Once the CLA is approved I'll run the standard tests. I have a couple updates and pull requests that I have in my backlog for this side project but will prioritize as soon as possible.

Might I ask you to open an enhancement issue and link the pull request with "Closes #x" when time permits.

**Comment by virtualistic** [comment]
> Thanks! Once the CLA is approved I'll run the standard tests. I have a couple updates and pull requests that I have in my backlog for this side project but will prioritize as soon as possible.
> 
> Might I ask you to open an enhancement issue and link the pull request with "Closes #x" when time permits.

Sure, no problem.   ( Enhancement request has a typo btw: "Please add a short _decsription_." )
When  I open a new Enhancement Request, there are several **mandatory** fields:
Description
Use Case(s)
Potential Configuration
Do I type:  "closes  # 670" in each field or do you want that placed in a certain field?

**Comment by tenthirtyam** [comment]
Closes is put here in the PR back to the enhancement issue.

thanks for the note on the issue form! :)

**Comment by virtualistic** [comment]
closes #671

**Comment by virtualistic** [comment]
> Closes is put here in the PR back to the enhancement issue.
> 
> thanks for the note on the issue form! :)

Most welcome!
Please note that the Oracle iso has an annoying mix of CaPs in its filename which I changed to all lowercase after I download the iso and also used the lowercase name in the packer file that references the iso.
So 2 questions:
1) Do you want me to use the original filename of the ISO in this pull request?

2) I also changed the iso_path to `iso/misc` for my situation in the `linux-oel.auto.pkrvars.hcl` . Do you want me to change it to `iso/linux/oracle` for this pull request?

**Comment by tenthirtyam** [comment]
Please use the original file name from the vendor and an oel or oracle path.

**Comment by virtualistic** [comment]
Added a new commit with the iso_path in line with the rest of the project and replaced the iso name with that of the vendor.

**Comment by tenthirtyam** [comment]
Would you consider including OEL9, too?

File: README.md [hunk 1]
```diff
@@ -843,6 +844,8 @@ The project supports using a datastore to store your guest operating system [`.i
      - [Download][download-linux-centos-7] the latest release of the **FULL** `.iso` image. (_e.g.,_ `CentOS-7-x86_64-DVD.iso`)
    - SUSE Linux Enterprise 15
      - [Download][download-linux-sles-15] the latest 15.4 release of the **FULL** `.iso` image. (_e.g.,_ `SLE-15-SP4-Full-x86_64-GM-Media1.iso`)
+   - Oracle Enterprise Linux 8
+     - [Download][download-linux-oel-8] the latest release of the **FULL** `.iso` image. (_e.g.,_ `oraclelinux-r8-u8-x86_64-dvd.iso`)
```
Comment (tenthirtyam) [inline README.md:848]:
```suggestion
     - [Download][download-linux-oel-8] the latest release of the **FULL** `.iso` image. (_e.g.,_ `OracleLinux-R8-U8-x86_64-dvd.iso`)
```

File: builds/linux/oracle/8/variables.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+}
+
+variable "vm_guest_os_family" {
+  type        = string
+  description = "The guest operating system family. Used for naming and VMware tools. (e.g. 'linux')"
+}
+
+variable "vm_guest_os_name" {
+  type        = string
+  description = "The guest operating system name. Used for naming . (e.g. 'rhel')"
```
Comment (tenthirtyam) [inline builds/linux/oracle/8/variables.pkr.hcl:87]:
```suggestion
  description = "The guest operating system name. Used for naming . (e.g. 'oel')"
```

File: builds/linux/oracle/8/variables.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+}
+
+variable "vm_guest_os_version" {
+  type        = string
+  description = "The guest operating system version. Used for naming. (e.g. '8')"
+}
+
+variable "vm_guest_os_type" {
+  type        = string
+  description = "The guest operating system type, also know as guestid. (e.g. 'rhel8_64Guest')"
```
Comment (tenthirtyam) [inline builds/linux/oracle/8/variables.pkr.hcl:97]:
```suggestion
  description = "The guest operating system type, also know as guestid. (e.g. 'oracleLinux8_64Guest')"
```

File: builds/linux/oracle/8/variables.pkr.hcl [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+// Removable Media Settings
+
+variable "common_iso_datastore" {
+  type        = string
+  description = "The name of the source vSphere datastore for ISO images. (e.g. 'sfo-w01-cl01-nfs01')"
+}
+
+variable "iso_path" {
+  type        = string
+  description = "The path on the source vSphere datastore for ISO image. (e.g. 'iso/linux/rhel')"
```
Comment (tenthirtyam) [inline builds/linux/oracle/8/variables.pkr.hcl:234]:
```suggestion
  description = "The path on the source vSphere datastore for ISO image. (e.g. 'iso/linux/oel')"
```

File: builds/linux/oracle/8/variables.pkr.hcl [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+}
+
+variable "iso_file" {
+  type        = string
+  description = "The file name of the ISO image used by the vendor. (e.g. 'rhel-<verssion>-x86_64-dvd.iso')"
```
Comment (tenthirtyam) [inline builds/linux/oracle/8/variables.pkr.hcl:239]:
```suggestion
  description = "The file name of the ISO image used by the vendor. (e.g. 'OracleLinux-<version>-x86_64-dvd.iso')"
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Minor changes.

Would you consider adding OEL 9?

**Comment by virtualistic** [comment]
Don't know if v9 will work because the latest vCenter v7 does **not**  offer Oracle Linux 9 as a guest os . 
(while my local ESXi server does)
2023-09-21 21_29_16-vCenter-oracleinux-versions

**Comment by virtualistic** [comment]
I agree with your suggested changes. Think it's best I edit my repo and create a new commit with the changes? 
Or should I do a "signoff and commit suggestion" from the GitHub UI?

**Comment by tenthirtyam** [comment]
> I agree with your suggested changes. Think it's best I edit my repo and create a new commit with the changes?
> Or should I do a "signoff and commit suggestion" from the GitHub UI?

 "signoff and commit suggestion" is fine, but I generally just amend the last commit and force push.

**Comment by tenthirtyam** [comment]
> Don't know if v9 will work because the latest vCenter v7 does **not** offer Oracle Linux 9 as a guest os . (while my local v7 ESXi server does).

Use `other5xLinux64Guest`. 😃

**Comment by virtualistic** [comment]
Ok, I changed all the requested files and did a :
```
git add -A
git commit --amend --no-edit
git push -f
```
Will have look if it's easy to add Oracle 9 now using my own Oracle 8 files :)

**Comment by virtualistic** [comment]
Ok, pushed a new commit with an Oracle 9 Template.
2023-09-22 13_56_08-oel9

**Comment by tenthirtyam** [comment]
Awesome! thank you!!

**Review by tenthirtyam — APPROVED** [review]
LGTM!

**Review by tenthirtyam — APPROVED** [review]
LGTM!

**Pull request merged**